internal/ocm/convert_test.go

```diff
@@ -566,3 +567,147 @@ func TestBuildCSExternalAuth(t *testing.T) {
 		})
 	}
 }
+
+func getBaseCSClusterBuilder(updating bool) *arohcpv1alpha1.ClusterBuilder {
+	var builder *arohcpv1alpha1.ClusterBuilder
+	clusterAPIBuilder := arohcpv1alpha1.NewClusterAPI()
+
+	if updating {
+		builder = arohcpv1alpha1.NewCluster()
+	} else {
+		// remove autoscaler as BuildCSCluster returns it separately
+		builder = ocmClusterDefaults().Autoscaler(nil)
+		clusterAPIBuilder = clusterAPIBuilder.Listening(arohcpv1alpha1.ListeningMethodExternal)
+	}
+
+	// Add common mutable fields that BuildCSCluster always sets
+	return builder.
+		NodeDrainGracePeriod(arohcpv1alpha1.NewValue().
+			Unit(csNodeDrainGracePeriodUnit).
+			Value(float64(0))).
+		API(clusterAPIBuilder.CIDRBlockAccess(arohcpv1alpha1.NewCIDRBlockAccess().
+			Allow(arohcpv1alpha1.NewCIDRBlockAllowAccess().
+				Mode(csCIDRBlockAllowAccessModeAllowAll))))
+}
+
+func TestBuildCSCluster(t *testing.T) {
+	testCases := []struct {
+		name              string
+		hcpCluster        *api.HCPOpenShiftCluster
+		updating          bool
+		expectedCSCluster *arohcpv1alpha1.ClusterBuilder
+	}{
+		{
+			name:     "CREATE - sets CIDRBlockAccess with nil AuthorizedCIDRs",
+			updating: false,
+			hcpCluster: &api.HCPOpenShiftCluster{
+				CustomerProperties: api.HCPOpenShiftClusterCustomerProperties{
+					API: api.CustomerAPIProfile{
+						AuthorizedCIDRs: nil,
+					},
+				},
+			},
+			expectedCSCluster: getBaseCSClusterBuilder(false),
+		},
+		{
+			name:     "CREATE - sets CIDRBlockAccess with empty AuthorizedCIDRs",
+			updating: false,
+			hcpCluster: &api.HCPOpenShiftCluster{
+				CustomerProperties: api.HCPOpenShiftClusterCustomerProperties{
+					API: api.CustomerAPIProfile{
+						AuthorizedCIDRs: []string{},
+					},
+				},
+			},
+			expectedCSCluster: getBaseCSClusterBuilder(false),
```

Comment:
Isn't an empty list of authorized CIDRS basically a request for blocking everyone (until this is updated with CIDR later)?

Reply:
> Isn't an empty list of authorized CIDRS basically a request for blocking everyone

No. the opposite - an empty list is "allow-all" by default. (similar to current behavior).